rn/Teacher/ios/Teacher/SpeedGrader/SubmissionCommentList.swift

```diff
@@ -0,0 +1,238 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2020-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+import Core
+
+struct SubmissionCommentList: View {
+    let assignment: Assignment
+    let submission: Submission
+    let filePicker = FilePicker()
+    @Binding var attempt: Int?
+    @Binding var fileID: String?
+    @Binding var showRecorder: MediaCommentType?
+
+    @Environment(\.appEnvironment) var env
+    @Environment(\.viewController) var controller
+
+    @ObservedObject var attempts: Store<LocalUseCase<Submission>>
+    @ObservedObject var comments: Store<GetSubmissionComments>
+
+    @State var comment: String = ""
+    @State var error: Text?
+    @State var showMediaOptions = false
+
+    init(
+        assignment: Assignment,
+        submission: Submission,
+        attempts: Store<LocalUseCase<Submission>>,
+        attempt: Binding<Int?>,
+        fileID: Binding<String?>,
+        showRecorder: Binding<MediaCommentType?>
+    ) {
+        self.assignment = assignment
+        self.submission = submission
+        self._attempt = attempt
+        self._fileID = fileID
+        self._showRecorder = showRecorder
+        self.attempts = attempts
+        comments = AppEnvironment.shared.subscribe(GetSubmissionComments(
+            context: .course(assignment.courseID),
+            assignmentID: assignment.id,
+            userID: submission.userID
+        ))
```

Comment:
submissions store in SpeedGraderView is loading everything. Which we may want to rethink later, because we could probably load each submission's full history only when we show that submission.